packages/tic-tac-toe/contracts/TicTacToe.sol

```diff
@@ -0,0 +1,617 @@
+        //     require(_new.bResolution() == _old.bResolution() - _new.stake());
+        // } else if (State.indexOfMover(_new) == 1) { // mover is B
+        //     require(_new.aResolution() == _old.aResolution() - _new.stake());
+        //     require(_new.bResolution() == _old.bResolution() + _new.stake());
+        // }
+
+    }
+
+    function requireValidXPLAYINGtoOPLAYING(
+        Outcome.AllocationItem[] memory fromAllocation,
+        Outcome.AllocationItem[] memory toAllocation,
+        TTTData memory fromGameData,
+        TTTData memory toGameData
+    )
+        private
+        pure
+        noDisjointMoves(toGameData)
+        stakeUnchanged(fromGameData, toGameData)
+        allocationsNotLessThanStake(fromAllocation, toAllocation, fromGameData, toGameData)
+    {
+        require(toGameData.Xs == fromGameData.Xs, 'No Xs added to board');
+        require(madeStrictlyOneMark(toGameData.Os, fromGameData.Os), 'One O placed');
+
+        // Current O Player should get all the stake. This is to decrease griefing. We assume that O Player is Player B
+        require(
+            toAllocation[0].amount == fromAllocation[0].amount.sub(toGameData.stake*2),
+            'Allocation for player A should be decremented by 1x stake'
+        );
+        require(
+            toAllocation[1].amount == fromAllocation[1].amount.add(toGameData.stake*2),
+            'Allocation for player B should be incremented by 1x stake.'
+        );
+
+        // Old TTTMagmo code
+        // if (State.indexOfMover(_new) == 0) {
+        //     // mover is A
+        //     require(_new.aResolution() == _old.aResolution() + 2 * _new.stake());
+        //     require(_new.bResolution() == _old.bResolution() - 2 * _new.stake());
+        // } else if (State.indexOfMover(_new) == 1) {
+        //     // mover is B
+        //     require(_new.aResolution() == _old.aResolution() - 2 * _new.stake());
+        //     require(_new.bResolution() == _old.bResolution() + 2 * _new.stake());
+        //     // note factor of 2 to swing fully to other player
+        // }
+
+    }
+
+    function requireValidOPLAYINGtoXPLAYING(
+        Outcome.AllocationItem[] memory fromAllocation,
+        Outcome.AllocationItem[] memory toAllocation,
+        TTTData memory fromGameData,
+        TTTData memory toGameData
+    )
+        private
+        pure
+        noDisjointMoves(toGameData)
+        stakeUnchanged(fromGameData, toGameData)
+        allocationsNotLessThanStake(fromAllocation, toAllocation, fromGameData, toGameData)
+    {
+        require(toGameData.Os == fromGameData.Os, 'No Os added to board');
+        require(madeStrictlyOneMark(toGameData.Xs, fromGameData.Xs), 'One X placed');
+
+        // Current X Player should get all the stake. This is to decrease griefing. We assume that X Player is Player A
+        require(
+            toAllocation[0].amount == fromAllocation[0].amount.add(toGameData.stake*2),
```

Comment:
Do you think we ought to use SafeMath throughout? E.g. `stake.mul(2)` instead of `stake*2`?